pkg/operator/ceph/pool/controller.go

```diff
@@ -165,32 +171,53 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 	// Populate clusterInfo during each reconcile
 	clusterInfo, _, _, err := mon.LoadClusterInfo(r.context, request.NamespacedName.Namespace)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
+		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to populate cluster info")
+	}
+	r.clusterInfo = clusterInfo
+
+	// Initialize the channel for this object store
+	// This allows us to track multiple CephBlockPool in the same namespace
+	_, ok := r.blockPoolChannels[cephBlockPool.Name]
```

Comment:
since `ok` is used so far below, how about making it more descriptive?
```suggestion
	_, poolChannelExists := r.blockPoolChannels[cephBlockPool.Name]
```